x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -0,0 +1,299 @@
+                          final ActionListener<EnrichPolicyResult> listener) {
+        // TODO: Index Pattern expansion and Alias support
+        // Collect the source index information
+        logger.info("Policy [{}]: Running enrich policy", policyName);
+        final String sourceIndexPattern = policy.getIndexPattern();
+        logger.debug("Policy [{}]: Checking source index [{}]", policyName, sourceIndexPattern);
+        GetIndexRequest getIndexRequest = new GetIndexRequest().indices(sourceIndexPattern);
+        client.admin().indices().getIndex(getIndexRequest, new ActionListener<GetIndexResponse>() {
+            @Override
+            public void onResponse(GetIndexResponse getIndexResponse) {
+                validateMappings(getIndexResponse, policyName, policy, listener);
+                prepareAndCreateEnrichIndex(policyName, policy, listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private Map<String, Object> getMappings(final GetIndexResponse getIndexResponse, final String sourceIndexName) {
+        ImmutableOpenMap<String, ImmutableOpenMap<String, MappingMetaData>> mappings = getIndexResponse.mappings();
+        ImmutableOpenMap<String, MappingMetaData> indexMapping = mappings.get(sourceIndexName);
+        // Assuming single mapping
+        MappingMetaData typeMapping = indexMapping.iterator().next().value;
+        return typeMapping.sourceAsMap();
+    }
+
+    private void validateMappings(final GetIndexResponse getIndexResponse, final String policyName, final EnrichPolicy policy,
+                                  final ActionListener<?> listener) {
+        String[] sourceIndices = getIndexResponse.getIndices();
+        logger.debug("Policy [{}]: Validating [{}] source mappings", policyName, sourceIndices);
+        for (String sourceIndex : sourceIndices) {
+            Map<String, Object> mapping = getMappings(getIndexResponse, sourceIndex);
+            @SuppressWarnings("unchecked")
+            Map<String, Object> properties = ((Map<String, Object>) mapping.get("properties"));
+            if (properties == null) {
+                listener.onFailure(
+                    new ElasticsearchException(
+                        "Enrich policy execution for [{}] failed. Could not read mapping for source [{}] included by pattern [{}]",
+                        policyName, sourceIndex, policy.getIndexPattern()));
+            }
+            if (properties.containsKey(policy.getEnrichKey()) == false) {
+                listener.onFailure(
+                    new ElasticsearchException(
+                        "Enrich policy execution for [{}] failed. Could not locate enrich key field [{}] on mapping for index [{}]",
+                        policyName, policy.getEnrichKey(), sourceIndex));
+            }
+            for (String enrichField : policy.getEnrichValues()) {
```

Comment:
I lean towards failing if a decorate field is missing in the mapping. I think this can lead to unexpected behaviour at enrich time? (why does this document not have all the enriched fields?)

Reply:
Even if the mapping exists in the source index, the document we are using to enrich may not have that field and will need handle partial decorations in the processor. 

If the decision to only allow full decorations, then the check here makes sense... but will need to changed slightly such that all enrich fields exist across the set of the source indexes (as opposed to exist in every source index as it is here). 

I am in favor of removing this check and implement `ignore_missing` in the processor to allow partial decorations.